cortex-mixin/config.libsonnet

```
@@ -38,6 +38,11 @@
       compactor: 'compactor.*',  // Match also custom compactor deployments.
     },
 
+    // Aggregation strings related to "jobs"
+    job_aggregation_prefix: 'cluster_namespace_job',
```

This can (and probably should) be auto-generated based on the value of `job_aggregation_labels_recording_rules`. It's about doing a string substitution:

```suggestion
    job_aggregation_prefix: std.strReplace(self. job_aggregation_labels_recording_rules, ', ', '_'),
```

---

I like this idea. Perhaps we can take it a step further, and define `_config.job_aggregation_labels` as a list, and that list is used to generate both the `","`-separated and `"_"`-separated strings; I assume those generated strings wouldn't need to appear in the config, only the list that they are generated from.

---

I simplified the _config aspect to only define the two lists that these prefix/group-by strings are generated from.
This required adding a new groups.libsonnet file. This helps with generating the strings consistently, and making them available along with the base _config.